modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,388 @@
+
+    forall(gen) { trusts =>
+      for {
+        store <- TrustStorage.make
+        firstTrust <- store.getTrust
+
+        _ <- store.updatePredictedTrust(trusts)
+
+        expected = trusts.view.mapValues(trustValue => TrustInfo(predictedTrust = trustValue.some)).toMap
+        actual <- store.getTrust.map(_.trust)
+      } yield
+        expect.all(
+          isEmpty(firstTrust),
+          expected === actual
+        )
+    }
+  }
+
+  test("internal trust storage is updated with public trust") {
+    val gen = for {
+      peerId <- peerIdGen
+      labels <- Gen.mapOfN(4, genPeerLabel)
+    } yield (peerId, PublicTrust(labels))
+
+    forall(gen) {
+      case (peerId, publicTrust) =>
+        for {
+          store <- TrustStorage.make
+          firstTrust <- store.getTrust
+
+          _ <- store.updatePeerPublicTrustInfo(peerId, publicTrust)
+
+          expected = PublicTrustMap(Map(peerId -> PublicTrust(publicTrust.labels)))
+          actual <- store.getTrust.map(_.peerLabels)
+        } yield
+          expect.all(
+            isEmpty(firstTrust),
+            expected === actual
+          )
+    }
+  }
+
+  test("initial current ordinal trust storage is empty") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getCurrentOrdinalTrust
+    } yield expect.all(OrdinalTrustMap.empty === trust)
+  }
+
+  test("initial next ordinal trust storage is empty except for the ordinal") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getNextOrdinalTrust
+    } yield
+      expect.all(
+        isEmpty(trust.trust),
+        SnapshotOrdinal(ordinalTrustUpdateInterval) === trust.ordinal,
+        PublicTrustMap.empty === trust.peerLabels
+      )
+  }
+
+  test("next ordinal trust storage doesn't update from internal storage because expected ordinal has not been reached") {
+    val minOrdinalValue = 0L
+    val maxOrdinalValue = ordinalTrustUpdateInterval - 1L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+          storedTrust <- store.getTrust
+
+          maybeOrdinalPublicTrust <- store.updateNext(ordinal)
+
+          next <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            trust === storedTrust,
+            Order.gt(SnapshotOrdinal(ordinalTrustUpdateInterval), ordinal),
+            maybeOrdinalPublicTrust.isEmpty,
+            isEmpty(next.trust),
+            SnapshotOrdinal(ordinalTrustUpdateInterval) === next.ordinal,
+            PublicTrustMap.empty === next.peerLabels
+          )
+    }
+  }
+
+  test("next ordinal trust storage doesn't update from internal storage because the internal storage is empty") {
```

Comment:
`because` -> `when the internal storage is empty`

Reply:
Changed to `next ordinal trust storage doesn't update from internal storage when the internal storage is empty`